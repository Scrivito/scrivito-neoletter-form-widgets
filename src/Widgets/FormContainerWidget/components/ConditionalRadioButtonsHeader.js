import * as React from "react";
import * as Scrivito from "scrivito";
import { getFieldName } from "../utils/getFieldName";
import { SelectItem } from "./SelectComponent";
import { Mandatory } from "./MandatoryComponent";
import { HelpText } from "./HelpTextComponent";

export const RadioButtonsHeader = Scrivito.connect(
  ({ widget, onChangeSelected }) => {
    return (
      <div
        className={`mb-3 select-container ${
          Scrivito.isInPlaceEditingActive() ? "condition-wrapper" : ""
        }`}
      >
        <div className="select-title">
          <span className="text-super">{widget.get("title")}</span>
          {widget.get("required") && <Mandatory />}
          {widget.get("helpText") && <HelpText widget={widget} />}
        </div>
        <div className="row ">
          {widget.get("conditions").map((condition, index) => (
            <SelectItem
              selectionType={"radio"}
              name={getFieldName(widget)}
              onChange={onChangeSelected}
              value={condition.get("title")}
              id={condition.id()}
              required={widget.get("required")}
              key={index}
            />
          ))}
        </div>
      </div>
    );
  }
);
