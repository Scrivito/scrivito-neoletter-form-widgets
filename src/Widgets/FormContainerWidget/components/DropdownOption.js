import * as React from "react";

export const DropdownOption = ({ value, id }) => {
  return (
    <option value={value} id={id}>
      {value}
    </option>
  );
};
