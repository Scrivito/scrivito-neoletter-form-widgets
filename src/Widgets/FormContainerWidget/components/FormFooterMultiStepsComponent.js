import * as React from "react";
import * as Scrivito from "scrivito";

export const FormFooterMultiSteps = Scrivito.connect(
  ({
    widget,
    onPageChange,
    onSubmit,
    currentStep,
    isLastPage,
    stepsLength,
  }) => {
    return (
      <div className="form-buttons">
        <button
          className="btn btn-primary backward-button"
          onClick={() => onPageChange(false)}
          hidden={currentStep == 1 && !Scrivito.isInPlaceEditingActive()}
        >
          {widget.get("backwardButtonText")}
        </button>
        <div className="step-counter">{currentStep + " / " + stepsLength}</div>
        <button
          className="btn btn-primary forward-button"
          onClick={isLastPage ? onSubmit : () => onPageChange(true)}
        >
          {isLastPage
            ? widget.get("submitButtonText")
            : widget.get("forwardButtonText")}
        </button>
      </div>
    );
  }
);
