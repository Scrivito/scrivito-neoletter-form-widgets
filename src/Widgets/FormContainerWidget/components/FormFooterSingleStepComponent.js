import * as React from "react";
import * as Scrivito from "scrivito";

export const FormFooterSingleStep = Scrivito.connect(({ widget, onSubmit }) => {
  return (
    <div
      className={`${
        widget.get("singleSubmitButtonAlignment") === "block"
          ? ""
          : widget.get("singleSubmitButtonAlignment")
      }`}
    >
      <button
        className={`btn btn-primary${
          widget.get("singleSubmitButtonAlignment") === "block"
            ? " btn-block"
            : ""
        }`}
        onClick={onSubmit}
      >
        {widget.get("submitButtonText")}
      </button>
    </div>
  );
});
