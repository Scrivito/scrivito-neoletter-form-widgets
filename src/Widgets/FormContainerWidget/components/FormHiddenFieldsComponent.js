import * as React from "react";
import * as Scrivito from "scrivito";
import { getHistory } from "../../../config/history";

export const FormHiddenFields = ({ widget }) => {
  const [browserLocation, setBrowserLocation] = React.useState(null);
  React.useEffect(() => {
    const history = getHistory();
    console.log("And the history is....");
    console.log(history);
    if (!history) return;
    setBrowserLocation(locationToUrl(history.location));

    return history.listen(({ location }) =>
      setBrowserLocation(locationToUrl(location))
    );
  }, []);
  return (
    <>
      <input type="hidden" name="form_id" value={widget.get("formId")} />
      <input
        type="hidden"
        name="url"
        value={browserLocation || Scrivito.urlFor(widget.obj())}
      />
      <Scrivito.InPlaceEditingOff>
        <Scrivito.ContentTag content={widget} attribute="hiddenFields" />
      </Scrivito.InPlaceEditingOff>

      <HoneypotField />
    </>
  );
};

const HoneypotField = () => (
  <div aria-hidden="true" className="winnie-the-pooh">
    <input autoComplete="off" name="fax" tabIndex="-1" />
  </div>
);

function locationToUrl(location) {
  return `${window.location.origin}${location.pathname}${location.search}`;
}
