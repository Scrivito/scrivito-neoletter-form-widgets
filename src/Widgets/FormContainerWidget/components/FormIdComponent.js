import * as React from "react";
import * as Scrivito from "scrivito";
import { ContentProperty } from "../../../Components/ContentProperty";
import { getScrivitoFormWidgetConfig } from "../../../config/scrivitoConfig";

export const FormIdComponent = Scrivito.connect(({ widget }) => {
  const formSubmissionsHref = widget.get("formId")
    ? `https://edit.neoletter.com/i/${
        getScrivitoFormWidgetConfig().tenant
      }/forms/${widget.get("formId")}`
    : null;
  const uiContext = Scrivito.uiContext();
  if (!uiContext) return null;

  return (
    <div className={`scrivito_${uiContext.theme}`}>
      <div className="scrivito_detail_content">
        <div className="attribute_form_id_item">
          <ContentProperty
            content={widget}
            attribute="formId"
            title="Form ID"
            description="This ID identifies the form in Neoletter."
          />

          <a
            className={`scrivito_button ${
              formSubmissionsHref ? "scrivito_blue" : "scrivito_disabled"
            }`}
            href={formSubmissionsHref}
            target="_blank"
            rel="noreferrer"
          >
            View submissions
          </a>
        </div>
      </div>
    </div>
  );
});
