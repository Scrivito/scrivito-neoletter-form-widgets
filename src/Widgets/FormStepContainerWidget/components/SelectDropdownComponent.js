import * as React from "react";
import { DropdownOption } from "./DropdownOption";

export const Dropdown = ({ options, name, id, required }) => {
  return (
    <select className="dropdown-select" name={name} id={id} required={required}>
      {<DropdownOption value={""} id={"empty-option"} key={"empty-option"} />}
      {options.map((optionValue, index) => (
        <DropdownOption value={optionValue} id={index} key={index} />
      ))}
    </select>
  );
};
