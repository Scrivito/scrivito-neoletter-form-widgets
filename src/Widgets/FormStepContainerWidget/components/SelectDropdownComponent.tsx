import * as React from "react";
import * as Scrivito from "scrivito";
import { DropdownOption } from "./DropdownOption";
import { Mandatory } from "./MandatoryComponent";
import { HelpText } from "./HelpTextComponent";

interface DropdownProps {
  options: string[];
  name: string;
  useFloatingLabel: boolean;
  widget: Scrivito.Widget;
}

export const Dropdown: React.FC<DropdownProps> = ({
  options,
  name,
  useFloatingLabel,
  widget
}) => {
  const id = widget.id();
  const required = widget.get("required") as boolean;
  const [isSelected, setIsSelected] = React.useState(false);


  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
    setIsSelected(event.target.value !== "");
  };

  return (
    <div className={`dropdown-wrapper ${useFloatingLabel ? 'floating-label' : ''} ${isSelected ? "is-selected" : ""}`}>
      <label htmlFor={id} className="dropdown-label">
        {widget.get("title") as string} {required && <Mandatory />}
        {widget.get("helpText") && <HelpText widget={widget} />}
      </label>
      <select
        className="dropdown-select"
        name={name}
        id={id}
        required={required}
        onChange={handleChange}
      >
        <DropdownOption value={""} id={"empty-option"} key={"empty-option"} />
        {options.map((optionValue, index) => (
          <DropdownOption value={optionValue} id={index.toString()} key={index} />
        ))}
      </select>
    </div>
  );
};
