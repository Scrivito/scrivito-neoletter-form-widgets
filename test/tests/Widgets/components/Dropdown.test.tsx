import React from "react";
import * as Scrivito from "scrivito";
import { fireEvent, render } from "@testing-library/react";
import { Dropdown } from "../../../../src/Widgets/FormStepContainerWidget/components/SelectDropdownComponent";
import { DummyWidget } from "../../../helpers/dummyWidget";

describe("Dropdown", () => {
  const widget = new DummyWidget({
    title: "Select some",
    helptext: "",
    required: true,
    customFieldName: "custom_dropdown"
  }) as unknown as Scrivito.Widget;

  const widgetProps = {
    options: ["Option 1", "Option 2", "Option 3"],
    name: "testDropdown",
    useFloatingLabel: false,
    widget: widget,
    required: true,
    helptext: "",
    title: "Select some",
    onInputChange: () => { }
  }
  it("renders the Dropdown component with options and empty dropdown", () => {

    const { container, getByText } = render(
      <Dropdown {...widgetProps} />
    );

    const emptyDropdown = container.querySelector("#empty-option");
    expect(emptyDropdown).toBeInTheDocument();

    widgetProps.options.forEach(option => {
      const optionElement = getByText(option);
      expect(optionElement).toBeInTheDocument();
    });
  });

  it("applies floating label class when useFloatingLabel is true", () => {

    const { container } = render(
      <Dropdown {...widgetProps} useFloatingLabel={true} />
    );

    const wrapper = container.querySelector(".dropdown-wrapper");
    expect(wrapper).toHaveClass("floating-label");
  });

  it("toggles is-selected class based on user selection", () => {

    const { container, getByRole } = render(
      <Dropdown  {...widgetProps} useFloatingLabel={true} />
    );

    const selectElement = getByRole("combobox");
    const wrapper = container.querySelector(".dropdown-wrapper");
    expect(wrapper).not.toHaveClass("is-selected");

    fireEvent.change(selectElement, { target: { value: "Option 1" } });
    expect(wrapper).toHaveClass("is-selected");
  });



  it("renders correctly", () => {

    const { container } = render(
      <Dropdown {...widgetProps} useFloatingLabel={false} />
    );

    const tree = container.firstChild;
    expect(tree).toMatchSnapshot();
  });
});
