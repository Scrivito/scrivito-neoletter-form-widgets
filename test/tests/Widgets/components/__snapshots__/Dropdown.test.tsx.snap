// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders correctly 1`] = `
<div
  class="dropdown-wrapper  "
>
  <label
    class="dropdown-label"
    for="fakeId"
  >
    <div
      class="mandatory-container"
    >
      <span
        class="text-mandatory"
      >
        *
      </span>
    </div>
  </label>
  <select
    class="dropdown-select form-select form-control"
    id="fakeId"
    name="testDropdown"
    required=""
  >
    <option
      id="empty-option"
      value=""
    />
    <option
      id="0"
      value="Option 1"
    >
      Option 1
    </option>
    <option
      id="1"
      value="Option 2"
    >
      Option 2
    </option>
    <option
      id="2"
      value="Option 3"
    >
      Option 3
    </option>
  </select>
</div>
`;
